diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4f25574f1..6f05134bf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -331,6 +331,7 @@ binaryen_add_executable(wasm-as src/tools/wasm-as.cpp)
 binaryen_add_executable(wasm-dis src/tools/wasm-dis.cpp)
 binaryen_add_executable(wasm-ctor-eval src/tools/wasm-ctor-eval.cpp)
 binaryen_add_executable(wasm-reduce src/tools/wasm-reduce.cpp)
+binaryen_add_executable(wasm-dwarf-dump src/tools/wasm-dwarf-dump.cpp)
 
 
 # binaryen.js
